docs/src/main/asciidoc/datastore.adoc

```diff
@@ -985,6 +985,8 @@ If a method's return type is `org.springframework.data.domain.Page`, the returne
 NOTE: Methods that return `Page` execute an additional query to compute total number of pages.
 Methods that return `Slice`, on the other hand, don't execute any additional queries and therefore are much more efficient.
 
+NOTE: You can enable nullability checks. For more details please see https://docs.spring.io/spring/docs/current/spring-framework-reference/core.html#null-safety[Spring Framework’s nullability docs].
```

Comment:
I think you deleted too much. The part about empty result handling was useful in my opinion. I would just collapse the whole nullability section into what you have here. The rest was fine.